sdk/src/Core/Strategies/DefaultStreamingStrategy.cs

```diff
@@ -0,0 +1,77 @@
+﻿using Amazon.XRay.Recorder.Core.Internal.Entities;
+using Amazon.XRay.Recorder.Core.Internal.Emitters;
+using Amazon.XRay.Recorder.Core.Sampling;
+
+namespace Amazon.XRay.Recorder.Core.Strategies
+{
+    public class DefaultStreamingStrategy : IStreamingStrategy
+    {
+        /// <summary>
+        /// Default max subsegment size to stream for the strategy.
+        /// </summary>
+        private const long DefaultMaxSubsegmentSize = 100;
+        
+        /// <summary>
+        /// Max subsegment size to stream fot the strategy.
+        /// </summary>
+        public long MaxSubsegmentSize { get; private set; } = DefaultMaxSubsegmentSize;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DefaultStreamingStrategy"/> class.
+        /// </summary>
+        public DefaultStreamingStrategy() : this(DefaultMaxSubsegmentSize)
+        {
+            
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DefaultStreamingStrategy"/> class.
+        /// </summary>
+        /// <param name="maxSubsegmentSize"></param>
+        public DefaultStreamingStrategy(long maxSubsegmentSize)
```

Comment:
We should do non-negative number check while setting this value. If negative number throw relevant exception. Also include unit test for the same.